test/test_context.py

```diff
@@ -44,8 +44,8 @@ class Key:
 
 @inject
 class Dependent:
-    some_singleton: SomeSingleton
-    some_component: SomeComponent
+    some_singleton = SomeSingleton
+    some_component = SomeComponent
```

Comment:
It would require a number of changes in `inject` and `context` to make this work. Moreover I'm not sure I like this approach because it requires you to write strange client code to accomodate the framework. Would it be reasonable to just not support `inject` as a class decorator in version 3.5 and below?